diff --git a/tripal_chado/includes/TripalFields/schema__alternate_name/schema__alternate_name_formatter.inc b/tripal_chado/includes/TripalFields/schema__alternate_name/schema__alternate_name_formatter.inc
index 58b08754..48d19a3e 100755
--- a/tripal_chado/includes/TripalFields/schema__alternate_name/schema__alternate_name_formatter.inc
+++ b/tripal_chado/includes/TripalFields/schema__alternate_name/schema__alternate_name_formatter.inc
@@ -15,6 +15,9 @@ class schema__alternate_name_formatter extends ChadoFieldFormatter {
     $list_items = array();
 
     foreach ($items as $delta => $item) {
+      if(empty($item['value'])){
+        continue;
+      }
       $list_items[] = $item['value'];
     }
     $list = 'There are no synonyms.';
diff --git a/tripal_chado/includes/TripalFields/schema__publication/schema__publication_formatter.inc b/tripal_chado/includes/TripalFields/schema__publication/schema__publication_formatter.inc
index 4b98c6a5..59ff9599 100755
--- a/tripal_chado/includes/TripalFields/schema__publication/schema__publication_formatter.inc
+++ b/tripal_chado/includes/TripalFields/schema__publication/schema__publication_formatter.inc
@@ -21,7 +21,9 @@ class schema__publication_formatter extends ChadoFieldFormatter {
     }
 
     foreach ($items as $delta => $item) {
-
+      if(empty($item['value'])){
+        continue;
+      }
       $title = isset($item['value']['TPUB:0000039']) ? $item['value']['TPUB:0000039'] : '';
       $citation = isset($item['value']['TPUB:0000003']) ? $item['value']['TPUB:0000003'] : '';
       $entity = (is_array($item['value']) && array_key_exists('entity', $item['value'])) ? $item['value']['entity'] : '';
