diff --git a/tripal_core/api/tripal_core.chado_query.api.inc b/tripal_core/api/tripal_core.chado_query.api.inc
index 6d285bb..13d356e 100644
--- a/tripal_core/api/tripal_core.chado_query.api.inc
+++ b/tripal_core/api/tripal_core.chado_query.api.inc
@@ -1091,8 +1091,6 @@ function chado_select_record($table, $columns, $values, $options = NULL) {
     return FALSE;
   }
 
-  $select = '';
-  $from = '';
   $where = array();
   $args = array();
 
@@ -1197,8 +1195,6 @@ function chado_select_record($table, $columns, $values, $options = NULL) {
       return array();
     }
 
-    $select[] = $field;
-
     // CASE 1: We have an array for a value.
     if (is_array($value)) {
 
